File: apps/cli/scripts/layouts/chat-assistant.lua
-- Portable declaration: the client renders this contract and executes its action.
function transition(state, event)
    if event.type == "schema" then
        return {state=state, payload={input={placeholder="Press /? for help"}, submit={operation="invoke"}}}
    end
    return {state=state, effects={}}
end

File: apps/cli/scripts/layouts/chat-assistant.ts
import type { PolicyEvent, PolicyResult } from "@inpainter/policy-runtime";

export function transition(state: unknown, event: PolicyEvent): PolicyResult {
  if (event.type === "schema") {
    return {
      state,
      payload: {
        input: { placeholder: "Press /? for help" },
        submit: { operation: "invoke" },
      },
    };
  }
  return { state, effects: [] };
}
